### PR TITLE
Last look client

### PR DESCRIPTION
**Pull request by gpxl-dev**
Implementing last look for the client as per [specification here](https://about.airswap.io/technology/last-look).

All feedback appreciated.

### CODE REVIEW

File: tools/protocols/src/Server.ts [hunk 1]
```diff
@@ -132,21 +159,168 @@ export class Server {
...
+  }
+
+  /**
+   * Throws if Last Look protocol is not supported, or if server is not yet
+   * initialized.
+   */
+  private requireLastLookSupport(version?: string) {
+    if (!this.supportsProtocol('last-look', version))
+      throw new Error(
+        `Server at ${this.locator} doesn't support this version of Last Look`
```
Comment (dmosites) [inline tools/protocols/src/Server.ts:320]: Possible to conditionally throw either "Server at ... doesn't support Last Look" or if version is set "Server at ... doesn't support this version of Last Look"?

File: tools/protocols/src/Server.ts [hunk 2]
```diff
@@ -1,69 +1,114 @@
...
+
+if (!isBrowser) {
+  JsonRpcWebsocket.setWebSocketFactory((url: string) => {
+    const ws = require('websocket').client
+    return new ws(url)
+  })
+}
+
+export class Server extends EventEmitter {
+  public transportProtocol: 'websockets' | 'http'
```
Comment (dmosites) [inline tools/protocols/src/Server.ts:50]: Suggest singular `websocket`

**Review by dmosites — APPROVED** [review]
Looks good! Suggest we update the package name to `@airswap/libraries` (start with version `0.0.1`) and rename its folder from `protocols` to `libraries` as well. Just a couple other comments as well.

**Review by hmalik88 — APPROVED** [review]
LGTM

**Pull request merged**